### ISSUE TITLE
application/o/authorize endpoint missing CORS headers

### ISSUE THREAD

**Issue by mdelpire** [goauthentik/authentik#10057]
**Describe the bug**
Not sure it is a bug or a misconfiguration somewhere...
Web application is calling the /application/o/authorize/ endpoint from the browser.

It is failing with the message in the Chrome Debug Console

`Access to fetch at 'https://auth.mydomain.com/application/o/authorize/?client_id=xxxxxxxxxx&redirect_uri=https%3A%2F%2Forigin.domain.com%2Foutpost.goauthentik.io%2Fcallback%3FX-authentik-auth-callback%3Dtrue&response_type=code&scope=email+profile+ak_proxy+openid&state=m2qJvqqq3tNbVgvzIgz7h6l3w4-hErhUX1fd5WmLJ_g' (redirected from 'https://origin.domain.com/info') from origin 'https://origin.domain.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request.`

I do not see the headers : access-control-allow-origin: https://origin.domain.com

**To Reproduce**

Create a Forward auth (Single Applicaiton) provider and set `External host` to `https://origin.domain.com`
Make sure the user was already logged in --> no new login requested.
Try to get authorisation from https://auth.mydomain.com/application/o/authorize/

**Expected behavior**
Preflight contains CORS headers 
`access-control-allow-origin: https://origin.domain.com`

**Version and Deployment (please complete the following information):**

-   authentik version: 2024.4.2
-   Deployment: docker-compose

**Comment by efaden** [comment]
Seeing this also

**Comment by Towerism** [comment]
This is probably because the preflight request is returning a 302 found instead of a 200 success.

**Comment by mdelpire** [comment]
It might be the case as if I clear the cache everything works fine. How can
I prevent this 302 and force to get the request?

**Comment by scipe** [comment]
Exact same issue. Fresh installation. K0s and ingress with nginx + load balancer + cloudflare

**Comment by PrivateGER** [comment]
Same issue. Makes Authentik proxy unusable on version 2024.8.2.

**Comment by QHivert** [comment]
Hello, same issue here.

Looking at the code, it seems that the authorize endpoint doesn't call the method [cors_allow](https://github.com/goauthentik/authentik/blob/main/authentik/providers/oauth2/utils.py#L33). So any OPTIONS request to that endpoint will fail.

I don't know if this is expected, though :/

**Comment by 0xEmma** [comment]
Also seeing this since update to 2024.8.2/2024.8.3

**Comment by yurividal** [comment]
Also facing this same issue

**Comment by Samppady** [comment]
I have exact the same issue.

**Comment by MahmoudAlyuDeen** [comment]
Same here, I suspect this is causing issues with properly getting `scopes` from authentik: `username`, `email`, `ak_proxy`, etc...

**Comment by dankozlowski** [comment]
Also running in to this on [2024.8.3](https://goauthentik.io/docs/releases/2024.8)

**Comment by StianHaugland1** [comment]
I have the same issue using nginx proxy manager on [2024.10.2](https://docs.goauthentik.io/docs/releases/2024.10)

**Comment by anultravioletaurora** [comment]
Getting this too trying to connect Owncloud Infinite Scale to Authentik 2024.10.1

**Comment by dankozlowski** [comment]
I was able to resolve this by verifying I had the correct redirect domains on my provider, for what it's worth.

**Comment by joriskt** [comment]
This is still an issue in 2024.12.0.

**Comment by Mdbook** [comment]
+1, have the same issue when trying to reverse proxy to this project https://github.com/will-moss/erin

**Comment by Pyramiden** [comment]
Same issue using Caddy as reverse proxy (for Immich)

**Comment by RomRider** [comment]
I'm running traefik in front of the outpost (embedded), adding something like this in the labels and assigning the middleware to the router fixes the issue for now:
```
traefik.http.middlewares.authentik-cors.headers.accessControlAllowOriginListRegex: ^https://.+\.(subdomain1|subdomain2)\.example\.org
```

**Comment by ssqlone** [comment]
+1 same issue any proofs?

**Comment by CodeAdminDe** [comment]
> Same issue using Caddy as reverse proxy (for Immich)

exactly the same issuer with ingress-nginx as reverse proxy.

**Comment by witalit** [comment]
I am also seeing this issue myself and have spent hours trying to troubleshoot. 

Anyone got anywhere as of yet?

**Comment by dasHimanshuSekhar** [comment]
can u guide me bro where you've passed data..

axios.post(
                `https://localhost:9443/application/o/prepaid-card-fino-isu/end-session/`,
                { redirect: "https://localhost:9443/api/v3/flows/executor/default-provider-invalidation-flow" },  // Request body
                {
                    params: { // Query parameters
                        id_token_hint: idTokenHint,
                        post_logout_redirect_uri: "http://localhost:3000/home",
                        state: ""
                    },
                    headers: {
                        "Content-Type": "application/json"
                    },
                    mode: 'cors'
                }
            )
            .then(() => {
                console.log("Logout successful");
                setUserInfo(null);
                setIdTokenHint(null); // Clear token hint
                window.location.href = "http://localhost:3000/home"; // Redirect to Home/Login
            })
            .catch(error => console.error("Logout Error:", error));


here is the code snipt

**Comment by mad73923** [comment]
I'm using traefik as forward proxy and solved this issue by setting the Content-Security-Policy Header of authentik:
```
traefik.http.middlewares.contentsec.headers.contentsecuritypolicy=frame-ancestors 'self' https://<URL to application>;
```
Documentation: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors

**Comment by dominic-r** [comment]
We're closing this in favor of this parent issue: #25263
